### PR TITLE
healenium changes and added in Beta Feature and Remote Web Driver Node installation fixed 

### PR DESCRIPTION
**Pull request by prashelke**
Thank you for your contribution.
Before submitting this PR, please make sure:

- [x] PR description and commit message should describe the changes done in this PR
- [x] Verify the PR is pointing to correct branch i.e. Release or Beta branch if the code fix is for specific release , else point it to master
- [x] Latest Code from master or specific release branch is merged to your branch
- [x] No unwanted\commented\junk code is included
- [x] No new warning upon build solution
- [x] Code Summary\Comments are added to my code which explains what my code is doing
- [ ] Existing unit test cases are passed
- [x] New Unit tests are added for your development
- [ ] Sanity Tests are successfully executed for New and Existing Functionality
- [x] Verify that changes are compatible with all relevant browsers and platforms.
- [x] After creating pull request there should not be any conflicts
- [x] Resolve all Codacy comments
- [ ] Builds and checks are passed before PR is sent for review
- [ ] Resolve code review comments
- [ ] Update the Help Library document to match any feature changes

## Summary by CodeRabbit

- **New Features**
  - Introduced Healenium functionality with checkboxes and URL input fields in the Agent Edit Page.
  - Added properties for Healenium self-healing and URL configuration.
  - Enhanced SeleniumDriver to support Healenium-based remote web driver initialization.

- **Bug Fixes**
  - Improved error handling and logging for missing browser driver files.

- **Tests**
  - Added unit tests for obtaining driver paths for different browsers using SeleniumDriver.

- **Refactor**
  - Updated feature descriptions and groupings related to Healenium.
  - Refactored file path handling for browser drivers in SeleniumRemoteWebDriverEditPage.

### CODE REVIEW

**Review by Maheshkale447 — APPROVED** [review]

File: Ginger/Ginger/Drivers/DriversConfigsEditPages/SeleniumRemoteWebDriverEditPage.xaml.cs [hunk 1]
```diff
@@ -195,37 +196,68 @@
...
-                System.IO.File.Copy(EdgeDriverFile, edgeDestFile);
+                
+                if(System.IO.File.Exists(edgeSourceFile))
+                {
+                    System.IO.File.Copy(edgeSourceFile, edgeDestFile);
+                }
+                else
+                {
+                    Reporter.ToLog(eLogLevel.DEBUG, "edge Driver not found");
+                }
```
[the remark this answers is not in the document]
Comment (prashelke) [reply Ginger/Ginger/Drivers/DriversConfigsEditPages/SeleniumRemoteWebDriverEditPage.xaml.cs]: added unit Test Case

File: Ginger/Ginger/Agents/AgentEditPage.xaml.cs [hunk 2]
```diff
@@ -101,6 +114,49 @@
...
+                    xHealeniumURLPnl.Visibility = Visibility.Collapsed;
+                }
+            }
+            else
+            {
+                xHealeniumcheckbox.IsEnabled = false;
+                xHealeniumcheckbox.Visibility = Visibility.Collapsed;
+                xHealeniumURLPnl.Visibility = Visibility.Collapsed;
+            }
+        }
```
[the remark this answers is not in the document]
Comment (prashelke) [reply Ginger/Ginger/Agents/AgentEditPage.xaml.cs:158]: will handled in next release

File: Ginger/Ginger/Agents/AgentEditPage.xaml.cs [hunk 3]
```diff
@@ -67,6 +68,18 @@ public AgentEditPage(Agent agent, bool isReadOnly = false, bool ignoreValidation
...
+                
+                if (WorkSpace.Instance.BetaFeatures.ShowHealenium)
+                {
+                    UpdateHealeniumUI();
+                }
+                else
+                {
+                    xHealeniumURLPnl.Visibility = Visibility.Collapsed;
+                    xHealeniumcheckbox.Visibility = Visibility.Collapsed;
+                }
```
[the remark this answers is not in the document]
Comment (prashelke) [reply Ginger/Ginger/Agents/AgentEditPage.xaml.cs:80]: will handled in next release

File: Ginger/Ginger/Drivers/DriversConfigsEditPages/SeleniumRemoteWebDriverEditPage.xaml.cs [hunk 4]
```diff
@@ -195,37 +196,68 @@
             string fireFoxDriverFile = "geckodriver.exe";
             string IEDriverFile = "IEDriverServer.exe";
             string EdgeDriverFile = "msedgedriver.exe";
-
-            string chromeSourceFile = System.IO.Path.Combine(assemblyLocation, chromeDriverFile);
+            SeleniumDriver seleniumDriver = new SeleniumDriver();
+            string chromeSourceFile = seleniumDriver.GetDriverPath(SeleniumDriver.eBrowserType.Chrome);
```
[the remark this answers is not in the document]
Comment (prashelke) [reply Ginger/Ginger/Drivers/DriversConfigsEditPages/SeleniumRemoteWebDriverEditPage.xaml.cs:200]: Updated

**Review by Maheshkale447 — APPROVED** [review]

**Pull request merged**